src/main/java/com/sublinks/sublinksapi/person/services/UserDataService.java

```diff
@@ -0,0 +1,47 @@
+package com.sublinks.sublinksapi.person.services;
+
+import com.sublinks.sublinksapi.person.config.UserDataConfig;
+import com.sublinks.sublinksapi.person.dto.Person;
+import com.sublinks.sublinksapi.person.dto.UserData;
+import com.sublinks.sublinksapi.person.repositories.UserDataRepository;
+import jakarta.annotation.Nullable;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+import java.util.Date;
+import java.util.Optional;
+
+@Component
+@RequiredArgsConstructor
+public class UserDataService {
+
+  private final UserDataRepository userDataRepository;
+  private final UserDataConfig userDataConfig;
+
+  public UserData save(UserData userData) {
+
+    return userDataRepository.save(userData);
```

Comment:
Please add events for this, it might be worth having later to watch for certain rules, actions, etc.

Please add events to the delete too.

Reply:
I also don't think there is a need to return the UserData, I believe all the other services return void.